tests/test_user_lock.rs

```diff
@@ -0,0 +1,63 @@
+use colink::{
+    extensions::instant_server::{InstantRegistry, InstantServer},
+    CoLink,
+};
+use std::{
+    sync::{Arc, Mutex},
+    thread,
+};
+
+async fn user_lock_fn(
+    cl: &CoLink,
+) -> Result<(), Box<dyn std::error::Error + Send + Sync + 'static>> {
+    cl.update_entry("example_lock_counter", &0_i32.to_le_bytes())
+        .await
+        .unwrap();
+    let test_num = 10;
+    let mut ths = vec![];
+    let ans = Arc::new(Mutex::new(vec![]));
+    for _ in 0..test_num {
+        let cl = cl.clone();
+        ths.push(thread::spawn({
+            let ans_clone = Arc::clone(&ans);
+            move || {
+                tokio::runtime::Builder::new_multi_thread()
+                    .enable_all()
```

Comment:
oh, just occur to me that shall we also assert that we have more than one thread running the test for this? (if not, we shall skip the test as it's not informative?)